base/src/main/java/org/aya/tyck/ExprTycker.java

```diff
@@ -155,6 +155,35 @@ public final class ExprTycker extends Tycker {
         var items = tuple.items().map(this::synthesize);
         yield new TermResult(new IntroTerm.Tuple(items.map(Result::wellTyped)), new FormTerm.Sigma(items.map(item -> new Term.Param(Constants.anonymous(), item.type(), true))));
       }
+      case Expr.CoeExpr coe -> {
+        assert coe.resolvedVar() instanceof DefVar<?, ?> defVar
+          && defVar.core instanceof PrimDef def && def.id == PrimDef.ID.COE : "desugar bug";
+        var defVar = coe.resolvedVar();
+        var mockApp = new Expr.AppExpr(coe.sourcePos(), new Expr.AppExpr(coe.sourcePos(),
+          new Expr.RefExpr(coe.id().sourcePos(), defVar),
+          new Expr.NamedArg(true, coe.type())),
+          new Expr.NamedArg(true, coe.restr()));
+        var res = synthesize(mockApp);
+        if (whnf(res.wellTyped()) instanceof PrimTerm.Coe(var type, var restr)) {
+          var bad = new Object() {
+            Term typeSubst;
+          };
+          var freezes = CofThy.conv(restr, new Subst(), subst -> {
+            var typeSubst = type.subst(subst);
+            // ^ `typeSubst` should now be instantiated under cofibration `restr`, and
+            // it must be the form of `(i : I) -> A`. We need to ensure the `i` does not occur in `A` at all
+            // by substituting `i` with any binding `?` and then find usage of `?` --- this makes life easier.
+            // See: https://github.com/ice1000/guest0x0/blob/main/base/src/main/java/org/aya/guest0x0/tyck/Elaborator.java#L293-L310
+            var yeah = new LocalVar("?");
+            var oh = ElimTerm.make(typeSubst, new Arg<>(new RefTerm(yeah), true));
+            int usages = oh.findUsages(yeah);
```

Comment:
Why not:

```kt
if (typeSubst is Lam lam) {
  lam.body.findUsages(lam.param)
}
else {
  GG directly
}
```

Reply:
Also add a case for pathlam

Reply:
I am not sure if the `typeSubst` can be a `PathLam` in the future.

Reply:
> Also add a case for pathlam

Okay